R/get_ref.R

```diff
@@ -438,6 +438,16 @@ get_cran_data <- function(package) {
   cran_current <- pkgcache::meta_cache_list(packages = package)[, c(
     "type", "package", "version", "published"
   )]
+  if (all(is.na(cran_current$published))) {
+    # workaround of https://github.com/r-lib/pkgcache/issues/109
+    db <- subset(tools::CRAN_package_db(), Package == package)
```

Comment:
This might be very slow as every run of the db call will download the whole `data.frame`.

How about we create a cache inside the package, as calling `tools::CRAN_package_db` every time will be very time consumingAnd initialize 

Add this outside the function (either on the same file or `utils.R`)
    - This sort of follows the same pattern as `pkgcache` in storing the cache on an internal `env`.

```r
pkgenv <- new.env(parent = emptyenv())
```

```suggestion
    if (is.null(pkgenv$cache_db)) {
      pkgenv$cache_db <- tools::CRAN_package_db()
    }
    db <- subset(pkgenv$cache_db, Package == package)

```

Reply:
thanks for a good suggestion. I have applied this